src/main/java/com/server/RequestParamsBuilder.java

```diff
@@ -0,0 +1,34 @@
+package com.server;
+
+import java.util.Hashtable;
+
+public class RequestParamsBuilder {
+    private String path;
+    private String method;
+    private Hashtable<String, String> queryComponent;
+    private Hashtable<String, String> cookies;
+
+    public RequestParamsBuilder setPath(String path) {
+        this.path = path;
+        return this;
+    }
+
+    public RequestParamsBuilder setMethod(String method) {
+        this.method = method;
+        return this;
+    }
+
+    public RequestParamsBuilder setQueryComponent(Hashtable<String, String> queryComponent) {
+        this.queryComponent = queryComponent;
+        return this;
+    }
+
+    public RequestParamsBuilder setCookies(Hashtable<String, String> cookies) {
+        this.cookies = cookies;
+        return this;
+    }
+
+    public RequestParams build() {
+        return new RequestParams(path, method, queryComponent, cookies);
+    }
+}
```

Comment:
Using Builder Pattern is such a great idea! I'm just wondering, how do you make sure that the required steps are being called? For example, setPath and setMethod have to be called to create a RequestParams, otherwise it will throw an error here.

Reply:
I'm not sure that it will throw an error if `setPath()` and `setMethod()` aren't called. But we would want it to, right? Maybe `path` and `method` could go into the constructor of the `Builder`?

Reply:
Or placing a check in the `build()` method that would raise an exception if the `path` and `method` properties haven't been set.